python/ray/tune/ray_trial_executor.py

```diff
@@ -166,8 +167,9 @@ def __init__(self,
 
         self._trial_cleanup = _TrialCleanup()
         self._has_cleaned_up_pgs = False
+        self._max_pending_trials = 1
```

Comment:
Can we maybe avoid adding more var if not necessary?
It's not used outside the scope of __init__ and set_max_pending_trials?

Semantic wise, is it the same as TrialRunner._max_pending_trials? I would avoid having similar information owned by different classes to avoid confusion of module responsibility.

Reply:
Good point, we can query `self._cached_actor_pg.maxlen` instead